lib/daru/view/adapters/googlecharts/base_chart.rb

```diff
@@ -8,12 +8,37 @@ class BaseChart
     # @return [Array, Daru::DataFrame, Daru::Vector, Daru::View::Table, String]
     #   Data of GoogleVisualr Chart
     attr_accessor :data
+    # @return [Hash] Various options created to facilitate more features.
+    #   These will be provided by the user
+    attr_accessor :user_options
 
     # @see #GoogleVisualr::DataTable.query_response_function_name
     def query_response_function_name(element_id)
       "handleQueryResponse_#{element_id.tr('-', '_')}"
     end
 
+    # @param data [Array, Daru::DataFrame, Daru::Vector, Daru::View::Table, String]
```

Comment:
All same methods (for chart and table) should be put into one module and include it into the classes where you want.